src/region.js

```diff
@@ -321,46 +247,13 @@ var Region = MarionetteObject.extend({
       // to throw a ViewDestroyedError for this view
       view._isDestroyed = true;
     }
+    if (shouldTriggerDetach) {
+      triggerMethodOn(view, 'detach', view);
+    }
     if (!view.supportsDestroyLifecycle) {
       triggerMethodOn(view, 'destroy', view);
     }
-  },
-
-  // Attach an existing view to the region. This
-  // will not call `render` or `onShow` for the new view,
-  // and will not replace the current HTML for the `el`
-  // of the region.
-  attachView: function(view) {
-    if (this.currentView) {
-      delete this.currentView._parent;
-    }
-    view._parent = this;
-    this.currentView = view;
-    return this;
-  },
-
-  // Checks whether a view is currently present within
-  // the region. Returns `true` if there is and `false` if
-  // no view is present.
-  hasView: function() {
-    return !!this.currentView;
-  },
-
-  // Reset the region by destroying any existing view and
-  // clearing out the cached `$el`. The next time a view
-  // is shown via this region, the region will re-query the
-  // DOM for the region's `el`.
-  reset: function() {
```

Comment:
Actually, that would be treating leading underscores as "package-private" when most of the time we mean "class private".  We may be calling leading underscore methods from across classes in other cases, but I'm not sure I like the potential for confusion.  Maybe just keep it public.